### PR TITLE
[14.0] storage_file: boost url compute

### PR DESCRIPTION
**Pull request by simahawk**
FWD port of https://github.com/OCA/storage/pull/88

### CODE REVIEW

**Review by mmequignon — APPROVED** [review]

File: storage_file/models/ir_actions_report.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import models
+
+
+class IrActionsReport(models.Model):
+    _inherit = "ir.actions.report"
+
+    def render_qweb_pdf(self, res_ids=None, data=None):
+        return super(
+            IrActionsReport, self.with_context(print_report_pdf=True)
+        ).render_qweb_pdf(res_ids=res_ids, data=data)
```
Comment (ivantodorovich) [inline storage_file/models/ir_actions_report.py:14]:
```suggestion
        self = self.with_context(print_report_pdf=True)
        return super().render_qweb_pdf(res_ids=res_ids, data=data)
```
Not sure what you think but I recently discovered this is the same and it's more readable.
Not blocking of course

File: storage_file/models/storage_backend.py [hunk 2]
```diff
@@ -79,3 +80,42 @@ def _get_backend_id_from_param(cls, env, param_name, default_fallback=True):
...
+        "directory_path",
+        "url_include_directory_path",
+    )
+    def _compute_base_url_for_files(self):
+        for record in self:
+            record.base_url_for_files = record._get_base_url_for_files()
+
+    def _get_base_url_for_files(self):
+        """Retrieve base URL for files."""
+        backend = self.sudo()
```
Comment (ivantodorovich) [inline storage_file/models/storage_backend.py:96]:
```suggestion
```

Is this necessary?
In this method compute is stored so it already runs on sudo, for the other cases maybe consider `compute_sudo=True`

**Review by ivantodorovich — APPROVED** [review]
not blocking of course
LG :+1:

**Comment by ivantodorovich** [comment]
@simahawk can we merge this?

**Pull request merged**